### PR TITLE
Enhancement in utilities

### PR DESCRIPTION
**Pull request by ethouris**
Extras:
* `Sprint` (one-argument version for C++03)
* `PrintableMod`: version of `Printable` for containers where each element must have a prefix.
* `FilterIf`: combined standard functions `remove_copy_if` and `transform`
* `forceDrift` for `DriftTracer`: required for information interchange between sockets
* Property macros. Shortcuts to define accessor methods bound to fields.

### CODE REVIEW

File: srtcore/utilities.h [hunk 1]
```diff
@@ -966,4 +1017,14 @@ inline ValueType avg_iir(ValueType old_value, ValueType new_value)
...
 }
 
+#define SRTU_PROPERTY_RR(type, name, field) type name() { return field; }
+#define SRTU_PROPERTY_RO(type, name, field) type name() const { return field; }
+#define SRTU_PROPERTY_WO(type, name, field) void name(type arg) { field = arg; }
+#define SRTU_PROPERTY_WO_CHAIN(otype, type, name, field) otype& name(type arg) { field = arg; return *this; }
+#define SRTU_PROPERTY_RW(type, name, field) SRTU_PROPERTY_RO(type, name, field); SRTU_PROPERTY_WO(type, name, field)
+#define SRTU_PROPERTY_RRW(type, name, field) SRTU_PROPERTY_RR(type, name, field); SRTU_PROPERTY_WO(type, name, field)
+#define SRTU_PROPERTY_RW_CHAIN(otype, type, name, field) SRTU_PROPERTY_RO(type, name, field); SRTU_PROPERTY_WO_CHAIN(otype, type, name, field)
+#define SRTU_PROPERTY_RRW_CHAIN(otype, type, name, field) SRTU_PROPERTY_RR(type, name, field); SRTU_PROPERTY_WO_CHAIN(otype, type, name, field)
```
Comment (maxsharabayko) [inline srtcore/utilities.h]: Add comments regarding the naming and usage instructions.

File: srtcore/utilities.h [hunk 2]
```diff
@@ -630,7 +636,15 @@ class UniquePtr: public std::auto_ptr<T>
     operator bool () { return 0!= get(); }
 };
 
-// A primitive one-argument version of Printable
+// A primitive one-argument versions of Sprint and Printable
+template <class Arg1>
+inline std::string Sprint(const Arg1& arg)
```
Comment (maxsharabayko) [inline srtcore/utilities.h]: SPrint might be better, to avoid confusion with "sprint"

Comment (ethouris) [reply srtcore/utilities.h]: It already exists. This is a C++03 version of an existing function. That's still not the best name of this function.

File: srtcore/utilities.h (hunk 1 above)
Comment (ethouris) [reply srtcore/utilities.h]: Right.

File: srtcore/utilities.h [hunk 3]
```diff
@@ -675,6 +689,37 @@ typename Map::mapped_type const* map_getp(const Map& m, const Key& key)
...
+    typedef typename Container::value_type Value;
+    std::ostringstream os;
+    os << "[ ";
+    for (typename Container::const_iterator y = in.begin(); y != in.end(); ++y)
+        os << prefix << Value(*y) << " ";
+    os << "]";
+    return os.str();
+}
+
+
```
Comment (rndi) [inline srtcore/utilities.h]: Please remove spurious line returns

File: srtcore/utilities.h [hunk 4]
```diff
@@ -966,4 +1017,48 @@ inline ValueType avg_iir(ValueType old_value, ValueType new_value)
...
+#define SRTU_PROPERTY_RR(type, name, field) type name() { return field; }
+#define SRTU_PROPERTY_RO(type, name, field) type name() const { return field; }
+#define SRTU_PROPERTY_WO(type, name, field) void name(type arg) { field = arg; }
+#define SRTU_PROPERTY_WO_CHAIN(otype, type, name, field) otype& name(type arg) { field = arg; return *this; }
+#define SRTU_PROPERTY_RW(type, name, field) SRTU_PROPERTY_RO(type, name, field); SRTU_PROPERTY_WO(type, name, field)
+#define SRTU_PROPERTY_RRW(type, name, field) SRTU_PROPERTY_RR(type, name, field); SRTU_PROPERTY_WO(type, name, field)
+#define SRTU_PROPERTY_RW_CHAIN(otype, type, name, field) SRTU_PROPERTY_RO(type, name, field); SRTU_PROPERTY_WO_CHAIN(otype, type, name, field)
+#define SRTU_PROPERTY_RRW_CHAIN(otype, type, name, field) SRTU_PROPERTY_RR(type, name, field); SRTU_PROPERTY_WO_CHAIN(otype, type, name, field)
+
```
Comment (rndi) [inline srtcore/utilities.h]: Please remove spurious line returns

**Pull request merged**